compiler/rustc_privacy/src/lib.rs

```diff
@@ -205,6 +211,10 @@ where
         let ty::GenericPredicates { parent: _, predicates } = predicates;
         predicates.iter().try_for_each(|&(predicate, _span)| self.visit_predicate(predicate))
     }
+
+    fn visit_clauses(&mut self, clauses: &[(ty::Clause<'tcx>, Span)]) -> ControlFlow<V::BreakTy> {
+        clauses.iter().try_for_each(|&(clause, _span)| self.visit_predicate(clause.as_predicate()))
```

Comment:
I think you should add a `visit_clause` and move all the `visit_predicate` logic there. Any non-clause `Predicate` will go into the `bug!` path in `visit_predicate` after all.